# Workflow and custom scripts of plastomes analysis of Medicago and its relatives

Extensive genomic rearrangements mediated by repetitive sequences in plastomes of Medicago and its relatives

## Authors

Shuang Wu, Jinyuan Chen, Ying Li, Ai Liu, Ao Li, Mou Yin, Nawal Shrestha, Jianquan Liu, Guangpeng Ren

## Contact information

wush19@lzu.edu.cn; rengp@lzu.edu.cn

## Content

### Assembly

[NOVOPlasty3.8.3](https://github.com/ndierckx/NOVOPlasty)
```
perl NOVOPlasty3.8.3.pl -c config.txt
```

### Annotation

[GeSeq](https://chlorobox.mpimp-golm.mpg.de/geseq.html)

[Geneious v.10.2.6](https://www.geneious.com/)

[OGDRAW v.1.3.1](https://chlorobox.mpimp-golm.mpg.de/OGDraw.html)
(The visual images of the annotations of all species were generated by OGDRAW v.1.3.1.)

## Inversion inference

[progressiveMauve v.2.4.0](http://darlinglab.org/mauve/user-guide/progressivemauve.html)

## Repeat analysis

### Repeat estimate

[Tandem Repeats Finder v.4.09](https://tandem.bu.edu/trf/trf.download.html)
```
/data/00/software/Repeatmask/Trf/trf409 merged-75-cp.fas 2 7 7 80 10 50 500 -f -d -m 
# Tandem Repeats Finder v.4.09 was used to characterize tandem repeats.
```

[REPuter](https://bibiserv.cebitec.uni-bielefeld.de/reputer;jsessionid=df4788eccb69fc26c76ffbad2a65)

We used REPuter program to identify dispersed repeats with a minimum repeat size of 30Â bp and identity of no less than 90% (hamming distance equal to 3).

### Caculate of repeat content
```
perl 05.Gff2maskGenome.pl repeat.gff individual.fasta mask_repeat.individual.fasta
# Mask repeat sequences with character "N" to eliminate the influence of nested or overlapping repeats in subsequent analysis.

perl 06.Count_repeat_content.pl mask_repeat.individual.fasta repeat_content.result.txt
# Caculate repeat content without the influence of nested or overlapping repeats.
```

## Comparative plastome analysis

### Display interspecific variations graphically
[mVISTA](https://genome.lbl.gov/vista/mvista/submit.shtml)

### Detect the sequence divergence and determine highly divergent regions

[get_annotated_regions_from_gb.py](https://github.com/Kinggerm/PersonalUtilities/)

[concatenate_fasta.py](https://github.com/Kinggerm/PersonalUtilities/)

```
python3 get_annotated_regions_from_gb.py merge.20taxa.gb -o result
mafft result/gene/gene.fasta > result/gene/gene-mafft/gene.mafft.fasta
mafft result/intergene/intergene.fasta > result/intergene/intergene-mafft/intergene.mafft.fasta
python3 concatenate_fasta.py gene_1.mafft.fasta gene_2.mafft.fasta ... gene_n.mafft.fasta -o PC.mafft.fasta
python3 concatenate_fasta.py intergene_1.mafft.fasta intergene_2.mafft.fasta ... intergene_n.mafft.fasta -o PN.mafft.fasta
python3 concatenate_fasta.py PC.mafft.fasta PN.mafft.fasta -o PCN.mafft.fasta
```

[MAFFT v.7.453](https://mafft.cbrc.jp/alignment/software/)

[DnaSP v.6](https://dnasp.software.informer.com/) (calculate the nucleotide diversity (pi) of each coding gene and non-coding regions (i.e. PC, PN and PCN))

## Phylogenetic inference
