# Workflow and custom scripts of plastomes analysis of Medicago and its relatives

Extensive genomic rearrangements mediated by repetitive sequences in plastomes of Medicago and its relatives

## Authors

Shuang Wu, Jinyuan Chen, Ying Li, Ai Liu, Ao Li, Mou Yin, Nawal Shrestha, Jianquan Liu, Guangpeng Ren

## Contact information

wush19@lzu.edu.cn; rengp@lzu.edu.cn

## Content

## Assembly

[NOVOPlasty3.8.3](https://github.com/ndierckx/NOVOPlasty)
```
perl NOVOPlasty3.8.3.pl -c config.txt
```

## Annotation

[GeSeq](https://chlorobox.mpimp-golm.mpg.de/geseq.html)

[Geneious v.10.2.6](https://www.geneious.com/)

[OGDRAW v.1.3.1](https://chlorobox.mpimp-golm.mpg.de/OGDraw.html)
(The visual images of the annotations of all species were generated by OGDRAW v.1.3.1.)

## Validation of introns

[trinityrnaseq-Trinity-v2.8.5](https://github.com/trinityrnaseq/trinityrnaseq/releases)

[TransDecoder-TransDecoder-v5.5.0](https://github.com/TransDecoder/TransDecoder/releases/tag/TransDecoder-v5.5.0)

[ncbi-blast-2.10.0+](https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/)

```
Trinity --seqType fq --single SRR1823822.fastq.gz --max_memory 60G --CPU 30 --full_cleanup --trimmomatic  --normalize_reads --output Mf_leaf1_trinity
Trinity --seqType fq --left SRR5115455_1.fastq.gz --right SRR5115455_2.fastq.gz --max_memory 60G --CPU 30 --full_cleanup --trimmomatic  --normalize_reads --output Ma2_trinity
Trinity --seqType fq --left SRR8281660_1.fastq.gz --right SRR8281660_2.fastq.gz --max_memory 60G --CPU 30 --full_cleanup --trimmomatic  --normalize_reads --output Tf7_trinity
TransDecoder-TransDecoder-v5.5.0/TransDecoder.LongOrfs -t trinity.result.fasta
ncbi-blast-2.10.0+/bin/makeblastdb -in trinity.result.CDS.fasta -input_type fasta -dbtype nucl -title name -out name_db
ncbi-blast-2.10.0+/bin/blastn -db name_db -query need_verify_gene.fasta -out blastn.result
```

## Inversion inference

[progressiveMauve v.2.4.0](http://darlinglab.org/mauve/user-guide/progressivemauve.html)

## Repeat analysis

### Repeat estimate

[Tandem Repeats Finder v.4.09](https://tandem.bu.edu/trf/trf.download.html)
```
/data/00/software/Repeatmask/Trf/trf409 merged-75-cp.fas 2 7 7 80 10 50 500 -f -d -m 
# Tandem Repeats Finder v.4.09 was used to characterize tandem repeats.
```

[REPuter](https://bibiserv.cebitec.uni-bielefeld.de/reputer;jsessionid=df4788eccb69fc26c76ffbad2a65)

We used REPuter program to identify dispersed repeats with a minimum repeat size of 30Â bp and identity of no less than 90% (hamming distance equal to 3).

### Caculate of repeat content
```
perl 05.Gff2maskGenome.pl repeat.gff individual.fasta mask_repeat.individual.fasta
# Mask repeat sequences with character "N" to eliminate the influence of nested or overlapping repeats in subsequent analysis.

perl 06.Count_repeat_content.pl mask_repeat.individual.fasta repeat_content.result.txt
# Caculate repeat content without the influence of nested or overlapping repeats.
```

## Comparative plastome analysis

### Display interspecific variations graphically
[mVISTA](https://genome.lbl.gov/vista/mvista/submit.shtml)

### Detect the sequence divergence and determine highly divergent regions

[get_annotated_regions_from_gb.py](https://github.com/Kinggerm/PersonalUtilities/)

[concatenate_fasta.py](https://github.com/Kinggerm/PersonalUtilities/)

```
python3 get_annotated_regions_from_gb.py merge.20taxa.gb -o result
mafft result/gene/gene.fasta > result/gene/gene-mafft/gene.mafft.fasta
mafft result/intergene/intergene.fasta > result/intergene/intergene-mafft/intergene.mafft.fasta
python3 concatenate_fasta.py gene_1.mafft.fasta gene_2.mafft.fasta ... gene_n.mafft.fasta -o PC.mafft.fasta
python3 concatenate_fasta.py intergene_1.mafft.fasta intergene_2.mafft.fasta ... intergene_n.mafft.fasta -o PN.mafft.fasta
python3 concatenate_fasta.py PC.mafft.fasta PN.mafft.fasta -o PCN.mafft.fasta
```

[MAFFT v.7.453](https://mafft.cbrc.jp/alignment/software/)

[DnaSP v.6](https://dnasp.software.informer.com/) (calculate the nucleotide diversity (pi) of each coding gene and non-coding regions (i.e. PC, PN and PCN))

## Phylogenetic inference

[concatenate_fasta.py](https://github.com/Kinggerm/PersonalUtilities/)

```
python3 concatenate_fasta.py shared_gene_1.mafft.fasta shared_gene_2.mafft.fasta ... shared_gene_n.mafft.fasta -o merged_shared_gene.mafft.fasta
clustalw2 -INFILE=merged_shared_gene.mafft.fasta -CONVERT -TYPE=DNA -OUTFILE=merged_shared_gene.mafft.phy -OUTPUT=PHYLIP
java -jar jmodeltest-2.1.7/jModelTest.jar -AIC -AICc -d merged_shared_gene.mafft.fasta -g 8 -o 00.jmodeltest.out -s 203 -tr 30 -f -i
```

[ClustalW v.2.1](http://www.clustal.org/download/current/)

[PhyML 3.1](https://github.com/stephaneguindon/phyml)

## Anlaysis of dN and dS substitution rates

[PAML4.8](http://abacus.gene.ucl.ac.uk/software/paml.html)

[RaxML](https://github.com/stamatak/standard-RAxML)

```
ynoo # calculate nonsynonymous (dN) and synonymous (dS) substitution rates 
RAxML/standard-RAxML-master/raxmlHPC-PTHREADS -s gene_mafft_nostop.phy -n gene_mafft_nostop -m GTRGAMMAI -f a -x 12345 -N 100 -p 12345 -T 30
# generate ML tree og gene using RAxML as constraint tree during the branch specific dN and dS rate estimation in PAML4.8/codeml free-ratio model (model=1)
codeml # branch specific dN and dS rate estimation in PAML4.8/codeml free-ratio model (model=1)
